lib/Runtime/Base/FunctionInfo.cpp

```diff
@@ -64,7 +64,7 @@ namespace Js
     FunctionBody *
     FunctionInfo::GetFunctionBody() const
     {
-        return functionBodyImpl == nullptr ? nullptr : functionBodyImpl->GetFunctionBody();
+        return (functionBodyImpl && functionBodyImpl->IsFunctionBody()) ? functionBodyImpl->GetFunctionBody() : nullptr;
```

Comment:
There is no harm in putting this additional check here as we already have an assert in FunctionProxy::GetFunctionBody but I think we should rather fix JsDiagGetFunctionPosition in either of 2 ways
1. If the function is not parsed, parse it and then return the information (preferable as user have no knowledge of defer-parse or re-deferal)
2. Check if function have body and if not return failure code